addon/components/g-map.js

```diff
@@ -30,11 +30,20 @@ export default Ember.Component.extend({
 
   didInsertElement() {
     this._super(...arguments);
-    if (isEmpty(this.get('map'))
+
+    let map = this.get('map');
+
+    if (isEmpty(map)
       && (typeof FastBoot === 'undefined')) {
       const canvas = this.$().find('.g-map-canvas').get(0);
       const options = this.get('permittedOptions');
-      this.set('map', new google.maps.Map(canvas, options));
+      map = new google.maps.Map(canvas, options);
+
+      google.maps.event.addListener(map, 'idle', () => {
```

Comment:
Can this event listener be removed after it fires the first time? Would that still solve your issue? I don't think we want every map triggering a resize every time it idles.